src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -176,7 +174,7 @@ private Type GetEquivalentType(Type type)
                     var interfacesWithInfo = info.ImplementedInterfaces.Select(i => new { Type = i, Info = i.GetTypeInfo() }).ToArray();
                     var singleTypeGenInterfacesWithGetType = interfacesWithInfo
                         .Where(i => i.Info.IsGenericType && i.Info.GenericTypeArguments.Length == 1)
-                        .Select(i => new { Type = i.Type, Info = i.Info, GenType = i.Info.GetGenericTypeDefinition() });
+                        .Select(i => new { i.Type, i.Info, GenType = i.Info.GetGenericTypeDefinition() }).ToList();
```

Comment:
I just looked at this code. Maybe we can get rid of the `Type` property in the anonymous type; we can use `AsType` on `TypeInfo` anyway to get back to it. This could get rid of a few anonymous type allocations higher up where we get `interfacesWithInfo`.